pkg/apis/operator/v1alpha1/tektonpipeline_defaults.go

```diff
@@ -150,6 +150,17 @@ func (p *Pipeline) setDefaults() {
 		p.EnableGitResolver = ptr.Bool(true)
 	}
 
+	// Statefulset Ordinals
+	// if StatefulSet Ordinals mode, buckets should be equal to replicas
+	if p.Performance.StatefulsetOrdinals == nil {
+		p.Performance.StatefulsetOrdinals = ptr.Bool(false)
+	} else if *p.Performance.StatefulsetOrdinals {
+		if p.Performance.Replicas != nil && *p.Performance.Replicas > 1 {
```

Comment:
If `p.Performance.Replicas != nil` then we might have to set the buckets to 1

Reply:
in statefulset mode, bucket should be equal to replicas, not to 1

Reply:
@jkhelil I guess @PuneetPunamiya try to say bucket should be `1`, when `p.Performance.Replicas == nil`

Reply:
yeah sorry for the confusion 
+1 on what @jkandasa said

Reply:
When replicas is not specified (it's nil), the default behavior of a StatefulSet is to create a single pod,  in StatefulSet's Knative mode, the default behavior is to create one bucket, so there is no need to enforce that mode, it works by default.

However, when replicas is explicitly set (not nil), I found that if we don’t enforce the bucket assignment to equal to replicas, the election mode and StatefulSet mode can interfere with each other. This results in some pods being in election mode while others are in StatefulSet mode, leading to inconsistent behavior.